docs/source/inference.rst

```diff
@@ -0,0 +1,189 @@
+  * **Embedding similarity ranking** Given a list of (e, e) enitity pairs or (r, r) relation pairs, calculate the similarity of for each pair, sort the resulting similarity score and output the topk most similar pairs.
+
+The ranking result will be automatically stored in the output file (result.tsv by default) using the tsv format.
+
+The Linkage Score Ranking
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The task of linkage score ranking is given a list of candidate (h, r, t) triplets, calculating the edge score of each triplet based on the trained model and the pretrained embeddings and then returning the topk most relevent triplets along with their scores. An example return value of top5 linkage score likes this::
+
+  src   relation dst   score (DistMult)
+  407   5        8429  3.5953474
+  3645  3        7121  3.585188
+  93    10       7035  3.4557137
+  93    9        7035  3.4197974
+  2441  5        4833  3.3639894
+
+DGL-KE provides dglke_score command to calculate linkage score ranking. Currently, we support six models in inference: TransE_l1, TransE_l2, RESCAL, DistMult, ComplEx, and RotatE.
+
+Four arguments are required to provide basic information for doning the linkage score ranking task:
+
+  * ``--data_path``, The path containing the id mapping files, include both the entity mapping file and the relation mapping file
+  * ``--model_path``, The path containing the pretrained model, include the embedding files (.npy) and a config.json containing the configure information of the model.
+  * ``--format``, The format of the input data, specified in ``h_r_t``. Ideally, user should provides three files, one for head entities, one for relations and one for tail entities. But we also allow users to use *** to represent *all* of the entities or relations. For example, ``h_r_*`` requires users to provide files containing head entities and relation entities and use the whole entity set as tail entities; ``*_*_t`` requires users to provide a single file containing tail entities and use the whole entity set as head entities and the whole relation set as relations. The supported formats include ``h_r_t``, ``h_r_*``, ``h_*_t``, ``*_r_t``, ``h_*_*``, ``*_r_*``, ``*_*_t``. By default, the calculation will take an N_h x N_r x N_t manner.
+  * ``--data_files`` A list of data file names. This is used to provide necessary files containing the requried data according to the format, e.g., for ``h_r_t``, three files are required as h_data, r_data and t_data, while for ``h_*_t``, two files are required as h_data and t_data.
+  * ``--raw_data``, A flag tells whether the data profiled in data_files is in the raw object naming space or in mapped id space. If True, the data is in the original naming space and the inference program will do the id translation according to id mapping files. If False, the data is just intergers and it is assumed that user has already done the id translation.
```

Comment:
It seems the arguments are also listed in the argument list file? can we organize these arguments like this (required arguments, task related arguments, etc) in the other file?

Reply:
Sure. We may discuss about how to present our arguments with a more readable way.